S7.Net/TPKT.cs

```diff
@@ -33,15 +33,25 @@ private TPKT(byte version, byte reserved1, int length, byte[] data)
         public static async Task<TPKT> ReadAsync(Stream stream, CancellationToken cancellationToken)
         {
             var buf = new byte[4];
-            int len = await stream.ReadExactAsync(buf, 0, 4, cancellationToken).ConfigureAwait(false);
+
+            Task<int> ansTask = stream.ReadExactAsync(buf, 0, 4, cancellationToken);
+            ansTask.Wait(stream.ReadTimeout);
+            if (!ansTask.IsCompleted) throw new TimeoutException();
+
+            int len = ansTask.Result;
             if (len < 4) throw new TPKTInvalidException("TPKT is incomplete / invalid");
 
             var version = buf[0];
             var reserved1 = buf[1];
             var length = buf[2] * 256 + buf[3]; //BigEndian
 
             var data = new byte[length - 4];
-            len = await stream.ReadExactAsync(data, 0, data.Length, cancellationToken).ConfigureAwait(false);
+
+            ansTask = stream.ReadExactAsync(data, 0, data.Length, cancellationToken);
+            ansTask.Wait(stream.ReadTimeout);
```

Comment:
As above

Reply:
ansTask.Wait(stream.ReadTimeout);  --> ansTask.Wait(cancellationToken);
maybe make it async code ...